core/src/test/java/be/xplore/pricescraper/services/ShoppingListServiceTests.java

```diff
@@ -2,48 +2,80 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
 
+import be.xplore.pricescraper.domain.shops.Item;
 import be.xplore.pricescraper.domain.users.ShoppingList;
 import be.xplore.pricescraper.domain.users.User;
 import be.xplore.pricescraper.exceptions.ShoppingListNotFoundException;
+import be.xplore.pricescraper.mocks.ShoppingListInMemoryRepository;
+import be.xplore.pricescraper.repositories.ItemPriceRepository;
+import be.xplore.pricescraper.repositories.ItemRepository;
+import be.xplore.pricescraper.repositories.ShopRepository;
+import be.xplore.pricescraper.repositories.ShoppingListRepository;
+import be.xplore.pricescraper.repositories.TrackedItemRepository;
+import be.xplore.pricescraper.repositories.UserRepository;
 import java.util.ArrayList;
+import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
 import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
-import org.springframework.context.annotation.Import;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
 import org.springframework.test.annotation.DirtiesContext;
-import org.springframework.test.context.jdbc.Sql;
 
 @DirtiesContext(classMode = DirtiesContext.ClassMode.BEFORE_EACH_TEST_METHOD)
-@DataJpaTest
-@Import({ShoppingListService.class, ItemService.class, UserService.class, ScraperService.class})
-@Sql({"/sql/shopping_list_test_data.sql"})
+@SpringBootTest(classes = {ShoppingListServiceImpl.class, ShoppingListInMemoryRepository.class})
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
 class ShoppingListServiceTests {
 
   @Autowired
-  ShoppingListService shoppingListService;
+  ShoppingListRepository shoppingListRepository;
   @Autowired
-  UserService userService;
+  ShoppingListService shoppingListService;
+
+  @MockBean
+  ItemRepository itemRepository;
+  @MockBean
+  ItemService itemService;
```

Comment:
Again, I would not mock services. These are parts of the domain that might contain essential logic that needs to be tested.